### PR TITLE
Screenshots gatherer and viewer

### PR DESCRIPTION
**Pull request by paulirish**
Just some WIP stuff. You'll see it's pretty straightforward though. 

I'm thinking we can surface this in the report. Will help to calibrate the FMP timestamps we're serving.

We do have a browser bug to sort out, though. For whatever reason these screenshots with emulation on have incorrect aspect ratio and include the gray area. This is a bug that dgozman will have to help us with.

### CODE REVIEW

File: cli/index.js [hunk 1]
```diff
@@ -50,6 +50,7 @@ Run Configuration:
     --load-page        Loads the page (default=true)
     --save-trace       Save the trace contents to disk
     --save-artifacts   Generate network dependency graph
+    --save-screenshots Oh yeah!
```
Comment (paullewis) [inline cli/index.js]: lllllol

File: src/gatherers/screenshots.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      return filmStripFrames.map((frame, i) => ({
+        timestamp: frame.timestamp,
+        datauri: images[i]
+      }));
+    });
+  }
+
+  postProfiling(options, tracingData) {
+    return this.getScreenshots(tracingData.traceContents).then(screenshots => {
+      if (options.flags.saveScreenshots) {
```
Comment (paullewis) [inline src/gatherers/screenshots.js]: This should live outside the gatherer. We already have code out for saving a trace file, it should live right next to that, and ideally the code that bakes the images should use Handlebars for consistency with the report.

**Comment by paullewis** [comment]
Reviewed. Overall it's tip-top, just want the save functionality out of the gatherer itself.

**Comment by paulirish** [comment]
The browser bug will be fixed by EOD: https://codereview.chromium.org/1980673003/

File: src/gatherers/screenshots.js (hunk 2 above)
Comment (brendankenny) [inline src/gatherers/screenshots.js]: +1

File: src/gatherers/screenshots.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+});
+</script>
+`;
+
+class ScreenshotFilmstrip extends Gather {
+
+  fetchScreenshot(frame) {
+    return frame
+      .imageDataPromise()
+      .then(data => Promise.resolve('data:image/jpg;base64,' + data));
```
Comment (brendankenny) [inline src/gatherers/screenshots.js]: can just be
`.then(data => 'data:image/jpg;base64,' + data);`

**Comment by paulirish** [comment]
Okay we're ready for a look here.

Logic moved alongside the --save-trace. In fact they're such buddies that I've merged them into a single flag. :)

Here's what the saved file looks like:

image


wdyt?

**Pull request merged**